### DISCUSSION TITLE
Self-hosted runner source code

### DISCUSSION THREAD

**Discussion by aw** [community/community#25580] [Actions]
<p>I was unable to find any links to the self-hosted runner’s source code. It would be nice if you can publish the sources and allow us to send in pull requests (ex: to support esoteric operating systems).</p>

**Comment by thboop** [comment, marked as answer]
<p>Hey <a class="mention" href="/aw1">@aw1</a> ,</p>
<p>We have a plan to open source the Actions Runner! Once it is available, we would love to review your pull requests. </p>

**Comment by pmatos** [comment]
<p>I also read that plan for the Runner to run on i386, arm32 and arm64 exists but I saw no mention of when it will happen. Any ideas?</p>
